api/src/main/java/org/pmiops/workbench/notebooks/LeonardoNotebooksClientImpl.java

```diff
@@ -97,39 +101,51 @@ private LeonardoCreateRuntimeRequest buildCreateRuntimeRequest(
-        .jupyterUserScriptUri(assetsBaseUrl + "/initialize_notebook_runtime.sh")
-        .jupyterStartUserScriptUri(assetsBaseUrl + "/start_notebook_runtime.sh")
-        .userJupyterExtensionConfig(
-            new LeonardoUserJupyterExtensionConfig().nbExtensions(nbExtensions))
-        // Matches Terra UI's scopes, see RW-3531 for rationale.
-        .addScopesItem("https://www.googleapis.com/auth/cloud-platform")
-        .addScopesItem("https://www.googleapis.com/auth/userinfo.email")
-        .addScopesItem("https://www.googleapis.com/auth/userinfo.profile")
-        .runtimeConfig(
-            new LeonardoMachineConfig()
-                .cloudService(CloudServiceEnum.DATAPROC)
-                .masterDiskSize(
-                    Optional.ofNullable(clusterOverride.masterDiskSize)
-                        .orElse(config.firecloud.notebookRuntimeDefaultDiskSizeGb))
-                .masterMachineType(
-                    Optional.ofNullable(clusterOverride.machineType)
-                        .orElse(config.firecloud.notebookRuntimeDefaultMachineType)))
-        .toolDockerImage(workbenchConfigProvider.get().firecloud.jupyterDockerImage)
-        .welderDockerImage(workbenchConfigProvider.get().firecloud.welderDockerImage)
-        .customEnvironmentVariables(customEnvironmentVariables);
+    LeonardoCreateRuntimeRequest request =
+        new LeonardoCreateRuntimeRequest()
+            .labels(
+                ImmutableMap.of(
+                    RUNTIME_LABEL_AOU,
+                    "true",
+                    RUNTIME_LABEL_CREATED_BY,
+                    userEmail,
+                    RUNTIME_LABEL_AOU_CONFIG,
+                    RuntimeConfigurationType.USEROVERRIDE.equals(runtime.getConfigurationType())
+                        ? "user-override"
+                        : "default"))
+            .defaultClientId(config.server.oauthClientId)
+            // Note: Filenames must be kept in sync with files in api/src/main/webapp/static.
+            .jupyterUserScriptUri(assetsBaseUrl + "/initialize_notebook_runtime.sh")
+            .jupyterStartUserScriptUri(assetsBaseUrl + "/start_notebook_runtime.sh")
+            .userJupyterExtensionConfig(
+                new LeonardoUserJupyterExtensionConfig().nbExtensions(nbExtensions))
+            // Matches Terra UI's scopes, see RW-3531 for rationale.
+            .addScopesItem("https://www.googleapis.com/auth/cloud-platform")
+            .addScopesItem("https://www.googleapis.com/auth/userinfo.email")
+            .addScopesItem("https://www.googleapis.com/auth/userinfo.profile")
+            .runtimeConfig(
+                new LeonardoMachineConfig()
+                    .cloudService(CloudServiceEnum.DATAPROC)
+                    .masterDiskSize(
+                        Optional.ofNullable(clusterOverride.masterDiskSize)
+                            .orElse(config.firecloud.notebookRuntimeDefaultDiskSizeGb))
```

Comment:
Sorry, this was not super clearly written in the ticket AC, but the main point of the ticket is to actually plumb through the values supplied by the client to create the runtime instance: machine type and disk size. That hasn't been done here. This should also only happen if the flag is enabled - otherwise we shouldn't expect to receive a runtime payload in the create request.

Reply:
where am I getting the machine type and disk size from? Do we have the UI for that already?

Reply:
We don't have it yet: https://precisionmedicineinitiative.atlassian.net/browse/RW-5410

Reply:
should I just send over the firecloud config defaults for now then? The values we're currently using in the backend call.